### DISCUSSION TITLE
Use Psalm to force deadlines on // TODO comments

### DISCUSSION THREAD

**Discussion by gdejong** [vimeo/psalm#10146] [Ideas]
Would it a an idea to be able to use Psalm to add a deadline to `// TODO` comments?

Imagine you as a developer write down a `// TODO fix this before 01/01/2024` comment. All to often these get forgotten about. 
Psalm could know about `// TODO` items that have a deadline, and report an error if the TODO is still present once that deadline has passed.

**Comment by weirdan** [comment]
I'd say it's out of scope, and you don't need anything Psalm provides for this. It probably easier done as a PHPCS rule.

**Comment by gdejong** [comment]
OK thanks for your reply!
